spec-cordova/prepare.spec.js

```diff
@@ -17,175 +17,265 @@
-            }).fin(done);
+
+        describe('success', function () {
+            beforeEach(function () {
+                spyOn(util, 'cdProjectRoot').and.returnValue(project_dir);
+                spyOn(util, 'preProcessOptions').and.callFake(function(options) {
+                    let platforms = options.platforms || [];
+                    return {'platforms':platforms};
+                });
+            });
+            it('should fire the before_prepare hook and provide platform and path information as arguments', function(done) {
+                prepare({}).then(function() {
+                    expect(HooksRunner.prototype.fire.calls.argsFor(0)[0]).toEqual('before_prepare');
+                    done();
+                });
+            });
+            it('should invoke restore module\'s installPlatformsFromConfigXML method', function(done) {
+                prepare({}).then(function() {
+                    expect(restore.installPlatformsFromConfigXML).toHaveBeenCalled();
+                    done();
+                }).fail(function(err){
+                    expect(err).toBeUndefined();
+                    done();
+                });
+            });
+            it('should retrieve PlatformApi instances for each platform provided', function(done) {
+                prepare({'platforms':['android', 'ios']}).then(function() {
+                    expect(platforms.getPlatformApi.calls.count()).toBe(4);
+                    expect(platforms.getPlatformApi.calls.argsFor(0)[0]).toBe('android');
+                    expect(platforms.getPlatformApi.calls.argsFor(0)[1]).toBe('/some/path/platforms/android');
+                    expect(platforms.getPlatformApi.calls.argsFor(1)[0]).toBe('ios');
+                    expect(platforms.getPlatformApi.calls.argsFor(1)[1]).toBe('/some/path/platforms/ios');
+                    expect(platforms.getPlatformApi.calls.argsFor(2)[0]).toBe('android');
+                    expect(platforms.getPlatformApi.calls.argsFor(3)[0]).toBe('ios');
+                    done();
+                }).fail(function(err){
+                    expect(err).toBeUndefined();
+                    done();
+                });
+            });
+            it('should invoke restore module\'s installPluginsFromConfigXML method', function(done) {
+                prepare({platforms:[]}).then(function() {
+                    expect(restore.installPluginsFromConfigXML).toHaveBeenCalled();
+                    done();
+                });
+            });
+            it('should invoke preparePlatforms method, providing the appropriate platforms', function(done) {
+                prepare({platforms:['android']}).then(function() {
+                    expect(prepare.preparePlatforms).toHaveBeenCalled();
+                    expect(prepare.preparePlatforms.calls.argsFor(0)[0][0]).toBe('android');
+                    expect(prepare.preparePlatforms.calls.argsFor(0)[1]).toBe('/some/path');
+                    done();
```

Comment:
Ditto, factor done into `.done(done);` and explicitly fail in the fail handler please :) 🎁